File: java/Arrays/11_Container_with_most_water.java
import java.util.Scanner;
class ContainerWithMostWater{
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length-1;
        int maxArea=0;
        int a = 0 ;
        while (l<r){
            a = Math.min(height[l],height[r]);
            maxArea= Math.max(a*(r-l),maxArea);
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }

        }
        return maxArea;
    }
    public static void main(String args[]){
        int heightSize;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of elements in the array : ");
        heightSize = sc.nextInt();
        int[] height= new int[heightSize];
        for (int i = 0; i < heightSize ; i ++){
            height[i]= sc.nextInt();
        }
        ContainerWithMostWater c = new ContainerWithMostWater();
        System.out.println("Container with most area is " + c.maxArea(height));
    }
}

File: c/Arrays/11_Container_with_most_water.c
#include<stdio.h>
int maxArea(int* height, int heightSize) {
    int l =0;
    int r= heightSize-1;
    int value=0;
    int a =0;
    while(l<r){
        a= (height[l]<height[r] ? height[l] : height[r])* (r-l);
        value = value>a ? value:a;

        if (height[l]<height[r]){
            l++;
        }
        else{
            r--;
        }
    }
    return value;
}

int main(){
    int heightSize;
    printf("Enter the number of elements : ");
    scanf("%d",&heightSize);
    int height[heightSize];

    for(int i =  0; i < heightSize ; i ++){
        scanf("%d",&height[i]);
    }
    printf("Most possible area is : %d",maxArea(height,heightSize));
    return 0;
}